<a href="https://colab.research.google.com/github/jaytrey777/Proect-2-Part-1/blob/main/Jason_Tracey_Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment: 

Choose and describe 2 potential datasets for your second project.
You will have a lot more freedom in your second project than in your first project. This is because we want you to have a project in your portfolio that interests you or relates to the industry you would like to work in.

Your task for this week is to propose two possible datasets you would like to work with for Project 2 and indicate which is your first and which is your second choice.  

You can choose datasets from either:

1. This list of pre-approved datasets and complete the tasks listed below for both of them
Pre-approved Datasets:
Stroke Prediction Dataset
Adult income dataset
Car Insurance Data
Metabolic Syndrome Prediction - dataset by informatics-edu
Pump It Up Challenge: Driven Data
Cirrhosis Prediction Dataset
Spanish Wine Quality Dataset


2. Other dataset(s) that align with an interest or field of expertise you already have.  

If you choose to source your own data, please be sure it meets the below requirements:
has a clear target or labels to predict
has potential for interesting analysis
the data must be available for use (it is your responsibility to ensure that the license states that you are able to use it.)
the data must be appropriate for a professional environment
the data must not contain personal information
the data must not be a dataset used for any assignment, lecture, or task from the course
the data must not be a time series dataset. You will be able to identify these because each row will represent a moment or interval of time. These kinds of datasets follow special rules and are not appropriate for the kind of machine learning you have learned in this stack.
is reasonable to work with during the time you have
A great dataset would also have:

8-25 columns (features)
1000 - 100,000 rows (instances)
both numeric and categorical features
At least some missing features in both categorical and numeric columns, so you can show off your data-cleaning skills!
Here are some resources to help you find great datasets:

The 50 Best Free Datasets for Machine Learning
21 Places to Find Free Datasets for Data Science Projects
Best Public Datasets for Machine Learning and Data Science
Top 20+ Datasets for Machine Learning and Statistics Projects
70+ Machine Learning Datasets & Project Ideas
DataRobot Pathfinder
You may choose a regression or classification problem for each proposed data set, but you might consider a classification project since you already have a regression project in your portfolio.  This way you can show your skills with both!

If you choose 2 datasets from outside of the pre-approved list, at least one of the datasets you choose must be approved by the instructor before you can use it for your project.

For this Task:
Whether you chose 2 pre-approved datasets or 2 datasets from another source, or one of each:

Create a notebook where you have uploaded and shown the .head() of each of your data sets. For each of the proposed datasets, include the following information in text cells:

First choice: dataset 1
1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person?  A business?  An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Second choice: dataset 2
1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person?  A business?  An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Create a NEW GitHub repository for this project, with a readme, and add your notebook to that new repository.  Submit the link below.

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
import pandas as pd

Data Loading

##1st Data Set

In [3]:
filename = '/content/drive/MyDrive/Colab Notebooks/02. Weeks 5-8: Machine Learning/07 Week 7: Classification/5. Core Assignments/nbagames.csv'
df = pd.read_csv(filename)
df.head()

GAME_DATE_EST   GAME_ID GAME_STATUS_TEXT  HOME_TEAM_ID  VISITOR_TEAM_ID  \
0    2022-12-22  22200477            Final    1610612740       1610612759   
1    2022-12-22  22200478            Final    1610612762       1610612764   
2    2022-12-21  22200466            Final    1610612739       1610612749   
3    2022-12-21  22200467            Final    1610612755       1610612765   
4    2022-12-21  22200468            Final    1610612737       1610612741   

   SEASON  TEAM_ID_home  PTS_home  FG_PCT_home  FT_PCT_home  ...  AST_home  \
0    2022    1610612740     126.0        0.484        0.926  ...      25.0   
1    2022    1610612762     120.0        0.488        0.952  ...      16.0   
2    2022    1610612739     114.0        0.482        0.786  ...      22.0   
3    2022    1610612755     113.0        0.441        0.909  ...      27.0   
4    2022    1610612737     108.0        0.429        1.000  ...      22.0   

   REB_home  TEAM_ID_away  PTS_away  FG_PCT_away  FT_PCT_away  FG3_PCT_away  \
0      46.0    1610612759     117.0        0.478        0.815         0.321   
1      40.0    1610612764     112.0        0.561        0.765         0.333   
2      37.0    1610612749     106.0        0.470        0.682         0.433   
3      49.0    1610612765      93.0        0.392        0.735         0.261   
4      47.0    1610612741     110.0        0.500        0.773         0.292   

   AST_away  REB_away  HOME_TEAM_WINS  
0      23.0      44.0               1  
1      20.0      37.0               1  
2      20.0      46.0               1  
3      15.0      46.0               1  
4      20.0      47.0               0  

[5 rows x 21 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26651 entries, 0 to 26650
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   GAME_DATE_EST     26651 non-null  object 
 1   GAME_ID           26651 non-null  int64  
 2   GAME_STATUS_TEXT  26651 non-null  object 
 3   HOME_TEAM_ID      26651 non-null  int64  
 4   VISITOR_TEAM_ID   26651 non-null  int64  
 5   SEASON            26651 non-null  int64  
 6   TEAM_ID_home      26651 non-null  int64  
 7   PTS_home          26552 non-null  float64
 8   FG_PCT_home       26552 non-null  float64
 9   FT_PCT_home       26552 non-null  float64
 10  FG3_PCT_home      26552 non-null  float64
 11  AST_home          26552 non-null  float64
 12  REB_home          26552 non-null  float64
 13  TEAM_ID_away      26651 non-null  int64  
 14  PTS_away          26552 non-null  float64
 15  FG_PCT_away       26552 non-null  float64
 16  FT_PCT_away       26552 non-null  float6

1. [Source of data](https://www.kaggle.com/datasets/nathanlauga/nba-games?resource=download&select=games.csv) 

2. Brief description of data
  - This dataset was collected to work on NBA games data. Uses nba stats website to create this dataset.

  - There are 5 datasets :

    - games.csv : all games from 2004 season to last update with the date, teams and some details like number of points, etc.
    - games_details.csv : details of games dataset, all statistics of players for a given game
    - players.csv : players details (name)
    - ranking.csv : ranking of NBA given a day (split into west and east on CONFERENCE column
    - teams.csv : all teams of NBA

3. What is the target?
  - Did the home team win or lose?

4. What does one row represent? 
  - One row represents the team playing that night.

5. Is this a classification or regression problem?
  - This is a classification problem.

6. How many features does the data have?
  - There are 20 features.

7. How many rows are in the dataset?
  - There are 26651 rows.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
  - The amount of data will make it challenging.  The cleaning should be pretty straight forward. 

# 2nd Data Set

In [5]:
filename = '/content/drive/MyDrive/Colab Notebooks/02. Weeks 5-8: Machine Learning/07 Week 7: Classification/5. Core Assignments/adult income dataset.csv'
df2 = pd.read_csv(filename)
df2.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [8]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


[Source of data](https://www.kaggle.com/datasets/wenruliu/adult-income-dataset)

2. Brief description of data
  - An individual’s annual income results from various factors. Intuitively, it is influenced by the individual’s education level, age, gender, occupation, and etc.

  - This is a widely cited KNN dataset. 

3. What is the target?
  - Is the income above $50k?

4. What does one row represent? 
  - One row represents the team playing that night.

5. Is this a classification or regression problem?
  - This is a classification problem.

6. How many features does the data have?
  - There are 14 features.

7. How many rows are in the dataset?
  - There are 48842 rows.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
  - There are a fair amount of object rows so they will need to be analyzed to determine the best way to interpret that data.